Plugin.pm

```diff
@@ -1208,8 +1245,25 @@ sub _trackItem {
 		$item->{line1} .= ' (' . cstring($client, 'PLUGIN_QOBUZ_HIRES') . ')';
 	}
 
-	if ( $track->{work} ) {
-		$item->{work} = $track->{work};
+	# Enhancements to work/composer display for classical music (tags returned from Qobuz are all over the place)
+	if ( $track->{album}->{isClassique} ) {
+		if ( $track->{work} ) {
+			$item->{work} = $track->{work};
+		} else {
+			# Try to extract the work from the title
+			if ( $track->{composer}->{name} && index($track->{composer}->{name}, (split ':\s*', $track->{title})[0]) == -1 ) {
+				$item->{work} = (split ':\s*', $track->{title})[0];
+			} else {
+				$item->{work} = (split ':\s*', $track->{title})[1];
+			}
+                        $item->{line1} =~ s/$item->{work}://;
+		}
+		$item->{displayWork} = $item->{work};
+		$item->{displayWork} = $track->{composer}->{name} . ": " . $item->{work} if ($track->{composer}->{name});
+		$item->{work} = $track->{composer}->{name} . ": " . $item->{work} if ($track->{composer}->{name});
+		my $composerSurname = (split ' ', $track->{composer}->{name})[-1];
```

Comment:
Again: don't access the result of the split unless you've verified there's a match. Otherwise you might end up with tons of warnings about undefined values in the log file.

Reply:
Done.